### DISCUSSION TITLE
Older BC Roaster - MODBUS Communication Issues on v. 3.0.2

### DISCUSSION THREAD

**Discussion by grey6erry** [artisan-roaster-scope/artisan#1719] [Q&A] [answered]
Hi All. I am roasting on an old-ish propane Buckeye Roaster BC2000 circa 2017-2018 and am having a MODBUS connection issue while on Artisan 3.0.2. The issue is that a connection to MODBUS is intermittent, resulting in a 'punctuated' graph for both of my probes. 

<img width="598" alt="image" src="https://github.com/user-attachments/assets/16ad841b-f928-42b2-8f6a-77d643f6a1c1">
<img width="701" alt="image (1)" src="https://github.com/user-attachments/assets/febf88c0-284f-4e9f-940d-bd354574b094">

The log returns the following:
> 10 21:34:18.571 Modbus Communication Resumed
> 9 21:34:16.720 Modbus Communication Error
> 8 21:34:14.709 Modbus Communication Error
> 7 21:34:12.568 Modbus Communication Resumed
> 6 21:34:10.712 Modbus Communication Error
> 5 21:34:09.089 Connected via MODBUS
> 4 21:34:08.943 Scope recording...
> 3 21:34:08.940 Scope monitoring...
> 2 21:34:08.825 Scope has been reset
> 1 21:34:06.507 Scope has been reset

I am using the latest FTDI driver and tested this on two Windows laptops with the equal results. The issue does not persist on a previous version of Artisan and MODBUS connection is rock solid. I noticed that there's been updates to the MODBUS in the newest Artisan version 3.0.2 and thought that I might submit this in case this might help.

Here's my system config:
> version = 3.0.2 (fbe0eeb)
> Architecture = ('64bit', 'WindowsPE')
> Machine = AMD64
> Platform name = Windows-11-10.0.22631-SP0
> Processor = AMD64 Family 23 Model 96 Stepping 1, AuthenticAMD
> Python Branch = tags/v3.12.3
> Python Build = ('tags/v3.12.3:f6650f9', 'Apr 9 2024 14:05:25')
> Python Compiler = MSC v.1938 64 bit (AMD64)
> Python Implementation = CPython
> Python Revision = f6650f9
> Python version = 3.12.3
> Release = 11
> System = Windows
> Version = 10.0.22631
> Win32 = ('11', '10.0.22631', 'SP0', 'Multiprocessor Free')

**Comment by MAKOMO** [comment, marked as answer]
Thanks for the detailed description of your issue. Could you please give the current [Continuous Build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous) a try and report back if this one behaves any better?

**Comment by grey6erry** [comment]
Can confirm - everything works nominally on 3.0.3. I did a factory reset after install just to be sure. Thank you for a rapid remedy! :)

![Screenshot 2024-10-20 213239](https://github.com/user-attachments/assets/b9f9e99f-a5f9-4a23-9353-ab1bb10095af)
